providers/providers.go

```diff
@@ -9,16 +9,25 @@ import (
 type Provider interface {
 	Data() *ProviderData
 	GetEmailAddress(*sessions.SessionState) (string, error)
+	GetUserDetails(*sessions.SessionState) (*UserDetails, error)
```

Comment:
Does this replace `GetEmailAddress`, it seems to from the PR but I'm not entirely sure?

Reply:
Well, actually that's the idea, so I guess I can remove GetEmailAddress completely.